File: kaftter/src/main/java/kaftter/service/TweetService.java
package kaftter.service;

import kaftter.converter.TweetEntityConverter;
import kaftter.repository.TweetRepository;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Slf4j
@Service
public class TweetService {
    private final TweetRepository tweetRepository;

    public TweetService(final TweetRepository tweetRepository) {
        this.tweetRepository = tweetRepository;
    }

    public void save(final kaftter.tweet.Tweet tweet) {
        final var tweetEntity = TweetEntityConverter.convert(tweet);
        tweetRepository.save(tweetEntity);
    }

}

File: kaftter-api/src/main/kotlin/kaftter/api/service/TweetService.kt
package kaftter.api.service

import kaftter.api.converter.convert
import kaftter.api.exception.UserNotFoundException
import kaftter.api.repository.SummarizedTweetRepository
import kaftter.api.repository.TweetRepository
import kaftter.api.vo.Tweet
import kaftter.api.vo.TweetSummary
import org.springframework.stereotype.Service

@Service
class TweetService(
        private val tweetRepository: TweetRepository,
        private val summarizedTweetRepository: SummarizedTweetRepository
) {
    fun save(tweet: Tweet) {
        tweetRepository.save(convert(tweet))
    }

    fun search(userId: Long): TweetSummary {
        val summarizedTweet = summarizedTweetRepository.findByUserId(userId)
        summarizedTweet?.let {
            return convert(it)
        }
        throw UserNotFoundException(userId)
    }

}
